### PR TITLE
Move ISBN checksum calculator to a separate class

### PR DESCRIPTION
**Pull request by IonBazan**
This PR adds more coverage to ISBN and EAN codes and moves ISBN checksum calculation to a separate class. EAN checksum has been removed from `Barcode` provider because it already existed in `Ean` calculator.
I have also improved `Inn` calculator to avoid unnecessary index offset.

### CODE REVIEW

File: test/Faker/Calculator/EanTest.php [hunk 1]
```diff
@@ -7,75 +7,76 @@
...
-        );
+        return [
+            ['1234567', 0],
+            ['2345678', 5],
+            ['3456789', 0],
+        ];
     }
 
-    public function ean8ValidationProvider()
+    public function ean8ValidationProvider(): array
```
Comment (bramceulemans) [inline test/Faker/Calculator/EanTest.php:19]: These are breaking changed which cannot happen in a minor release. If someone extends the function, and has a different return type.

**Comment by bramceulemans** [comment]
Thanks for the contribution,however I see there are some breaking
- Adding types to non-final classes can break extended classes

File: test/Faker/Calculator/EanTest.php (hunk 1 above)
Comment (IonBazan) [reply test/Faker/Calculator/EanTest.php:19]: Does BC promise apply to the test classes too?

Comment (bramceulemans) [reply test/Faker/Calculator/EanTest.php:19]: Oops. I didn't realise it was a test class. Sorry!

Comment (pimjansen) [reply test/Faker/Calculator/EanTest.php:19]: Lets not do a BC change at all at this point

**Comment by pimjansen** [comment]
@bramceulemans can you review this as i dont have access to a computer and the Github app is pretty bad for this so.

**Comment by bramceulemans** [comment]
Will do when I get home from work.

**Review by bramceulemans — APPROVED** [review]
Looks good to me.﻿

**Pull request merged**